packages/ui/src/composables/useTrapFocus.ts

```diff
@@ -0,0 +1,92 @@
+import { nextTick, Ref } from 'vue'
+import { useDocument } from './useDocument'
+import { useWindow } from './useWindow'
+
+const TAB_KEYCODE = 9
+const FOCUSABLE_ELEMENTS_SELECTOR = ':where(a, button, input, textarea, select):not([disabled]), *[tabindex]'
+let trapInEl: HTMLElement | null = null
+
+export const useTrapFocus = () => {
+  const document = useDocument()
+  const window = useWindow()
+
+  let focusableElements: HTMLElement[] = []
+  let firstFocusableElement: HTMLElement | null = null
+  let lastFocusableElement: HTMLElement | null = null
+  let isFocusTrapped = false
+
+  const focusFirstElement = () => {
+    if (firstFocusableElement) {
+      firstFocusableElement.focus()
+    }
+  }
+  const focusLastElement = () => {
+    if (lastFocusableElement) {
+      lastFocusableElement.focus()
+    }
+  }
+
+  const onKeydown = (evt: KeyboardEvent) => {
+    const isTabPressed = evt.key === String(TAB_KEYCODE) || evt.keyCode === TAB_KEYCODE
+    const isShiftPressed = evt.shiftKey
+
+    if (!isTabPressed) {
+      return
+    }
+
+    if (!isFocusTrapped) {
+      isFocusTrapped = true
+
+      evt.preventDefault()
+      isShiftPressed ? focusLastElement() : focusFirstElement()
+
+      return
+    }
+
+    if (document.value?.activeElement === lastFocusableElement && !isShiftPressed) {
+      evt.preventDefault()
+      focusFirstElement()
+      return
+    }
+
+    if (document.value?.activeElement === firstFocusableElement && isShiftPressed) {
+      evt.preventDefault()
+      focusLastElement()
+    }
+  }
+
+  const trapFocusIn = (el: HTMLElement) => {
+    trapInEl = el
+
+    freeFocus()
+    trapFocus()
+  }
+
+  const trapFocus = () => {
+    nextTick(() => {
```

Comment:
Let's call `nextTick` outside `trapFocus` method e.g. `nextTick(trapFocus)`. 

BTW, why do we need `nextTick`?

Reply:
You are right with `nextTick` outside `trapFocus` 

We need `nextTick` because `trapFocus` works with DOM, so we must be sure that DOM has been updated after modal opened